packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -2,23 +2,70 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 import 'dart:async';
+import 'dart:convert';
 
 import 'package:pedantic/pedantic.dart';
 import 'package:vm_service/vm_service.dart';
 
 import '../config_specific/logger.dart';
 import '../globals.dart';
+import '../ui/fake_flutter/fake_flutter.dart';
 import '../vm_service_wrapper.dart';
+
 import 'memory_protocol.dart';
 import 'memory_service.dart';
 
+// TODO(terry): Temporary code for canned data.
+typedef chartStateListener = void Function();
+
+// TODO(terry): Implement a dispose method and call in ProvidedControllers dispose.
 /// This class contains the business logic for [memory.dart].
 ///
 /// This class must not have direct dependencies on dart:html. This allows tests
 /// of the complicated logic in this class to run on the VM and will help
 /// simplify porting this code to work with Flutter Web.
 class MemoryController {
-  MemoryController();
+  final MemoryTimeline memoryTimeline = MemoryTimeline();
+
+  // TODO(terry): Temporary code for canned data.
+  // *************************************************************************
+  // *************************************************************************
```

Comment:
why do we still need canned data if the live data is hooked up?

Reply:
Its gone.